include/node.tcc

```diff
@@ -518,3 +528,23 @@ bool mpm::Node<Tdim, Tdof, Tnphases>::mpi_rank(unsigned rank) {
   auto status = this->mpi_ranks_.insert(rank);
   return status.second;
 }
+
+//! Update nodal property at the nodes from particle
+template <unsigned Tdim, unsigned Tdof, unsigned Tnphases>
+void mpm::Node<Tdim, Tdof, Tnphases>::update_property(
+    bool update, const std::string& property,
+    const Eigen::MatrixXd& property_value, unsigned mat_id,
+    unsigned nprops) noexcept {
+  // Decide to update or assign
+  const double factor = (update == true) ? 1. : 0.;
+
+  // Calculate updated property
+  Eigen::MatrixXd updated_property =
+      property_handle_->property(property, prop_id_, mat_id, nprops) +
+      factor * property_value;
+
+  // Update/assign property
+  std::lock_guard<std::mutex> guard(node_mutex_);
+  property_handle_->assign_property(property, prop_id_, mat_id,
```

Comment:
Shouldn't the property_handle's assign_property update rather than node retrieving and updating the value?

Reply:
Is it returning a node? I thought my implementation was just assigning the property to the location associated to the node